src/components/PageHarvest.js

```diff
@@ -28,6 +28,7 @@ class PageHarvest extends Component {
 
   harvestHandler(spec) {
     const { dispatch, token, queue } = this.props
+    dispatch(uiNotificationNew({ type: 'info', message: 'Harvesting started.', timeout: 5000 }))
```

Comment:
Not required up front (this is a huge improvement) but, ...  Any thought on how to get this going away when the response comes (as opposed to just timing out), and reporting errors?

Reply:
The notifications system works well with multiple alerts. So the *easiest* thing would be to just add new error/complete alerts (they appear below the harvesting one, which would then go away, and the error one below it slides up in it's place):
![image](https://user-images.githubusercontent.com/299129/36866012-0246df8a-1d46-11e8-9087-479ccf68c357.png)

Having said that, each notification gets an id that is a part of the action's return object. So all you need to do in those cases is store that id. Then you can delete or update that message in the redux state. So maybe in the harvester reducer, look our for the `UI_NOTIFICATION_NEW` action where the message is 'Harvesting started' (we may want to store the messages as constants), then store that message ID as part of the current harvest request. When the harvest is complete, check if the id is still in the notifications list, if so you can update it or delete it.